docs/_assets/js/api-search.js

```diff
@@ -28,66 +28,44 @@
  *  }
  *  ```
  */
-onmessage = function(e) {
-    var docs = e.data.docs;
-    var searchTerm = e.data.search;
-
-    var regexForTerm = function(query) {
-        var escaped = query.replace(/([\.\*\+\?\|\(\)\[\]\\])/g, '\\$1');
-        if (query.toLowerCase() != query) {
-            // Regexp that matches CamelCase subbits: "BiSe" is
-            // "[a-z]*Bi[a-z]*Se" and matches "BitSet", "ABitSet", ...
-            return new RegExp(escaped.replace(/([A-Z])/g,"[a-z]*$1"));
-        }
-        else { // if query is all lower case make a normal case insensitive search
-            return new RegExp(escaped, "i");
-        }
+onmessage = function({ data: { docs, search } }) {
+    const regexForTerm = (query) => {
+        const escaped = query.replace(/([\.\*\+\?\|\(\)\[\]\\])/g, '\\$1');
+        return query.toLowerCase() !== query
+            ? new RegExp(escaped.replace(/([A-Z])/g, "[a-z]*$1"))
+            : new RegExp(escaped, "i");
```

Comment:
I am not a fan of this syntax (I find it more difficult to read, the previous one is more similar to what we have in scala, and it adds some inconsistency to the code, with the other ifs being unchanged), and I doubt it brings any performance improvements. Could you use the previous one?

Reply:
I'm a little biased (I wrote the change, so there is bias), but I find it easier to read. To me, ternary allows for more code in less space, and it keeps the statements concise and organized. If you think that an `if/then` statement is better, I'll change it back, but I prefer ternary.

Reply:
Let's at least bring back both of the comments that were in the if else before

Reply:
Right! If we are adding the comments, then you are right and the if/then may be prettier.